micro-rdk/src/common/config_monitor.rs

```diff
@@ -43,7 +69,15 @@ impl<'a> PeriodicAppClientTask for ConfigMonitor<'a> {
                 app_client.clone().get_config(None).await.unwrap();
             match self.curr_config == *new_config {
                 true => Ok(Some(self.get_default_period())),
-                false => self.restart(),
+                false => {
+                    if let Err(e) = self
+                        .storage
+                        .store_robot_configuration(new_config.config.unwrap())
```

Comment:
I'm suspicious of `unwrap`. What should happen here if `new_config.config` is None?

Reply:
not sure, what happens in entry.rs when get_config is called and new_config is None or returns an AppClientError?

https://github.com/viamrobotics/micro-rdk/blob/f95d10b20e2970c1f32b203ff4f7e1cb4932521e/micro-rdk/src/esp32/entry.rs#L83

Reply:
I'm guessing it panics?

Reply:
oh I see, I already new_config is already unwrapped